File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

lst = []
line = int(input())

for i in range(line):
    inp = input().split()
    if len(inp)==2:
        command = inp[0]
        num = int(inp[1])
    else:
        command = inp[0]

    if command=='push':
        lst.append(num)
    elif command=='top':
        if len(lst)>0:
            print(lst[-1])
        else:
            print(-1)
    elif command=='size':
        print(len(lst))
    elif command=='empty':
        if len(lst)==0:
            print(1)
        else:
            print(0)
    elif command=='pop':
        if len(lst)>0:
            print(lst[-1])
            del lst[-1]
        else:
            print(-1)

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int T = Integer.parseInt(br.readLine());
        Stack<Integer> st = new Stack<>();
        for(int i=0;i<T;i++){
            String[] input = br.readLine().split(" ");
            String cmd = input[0];
            if(cmd.equals("push")){
                st.add(Integer.parseInt(input[1]));
            }
            else if(cmd.equals("pop")){
                if(!st.isEmpty()){
                    System.out.println(st.pop());
                }
                else{
                    System.out.println(-1);
                }

            }
            else if(cmd.equals("size")){
                System.out.println(st.size());
            }
            else if(cmd.equals("empty")){
                if(st.isEmpty()){
                    System.out.println(1);
                }
                else{
                    System.out.println(0);
                }
            }
            else if(cmd.equals("top")){
                if(!st.isEmpty()){
                    System.out.println(st.peek());
                }
                else{
                    System.out.println(-1);
                }
            }
        }
    }
}
